helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -781,8 +841,15 @@ void reset(boolean isShutdown) {
           if (isShutdown) {
             _batchCallbackProcessor.shutdown();
             _batchCallbackProcessor = null;
+            if (_periodicRefreshExecutor != null) {
+              _periodicRefreshExecutor.shutdownNow();
+            }
           } else {
             _batchCallbackProcessor.resetEventQueue();
+            if (_periodicRefreshExecutor != null) {
+              _periodicRefreshExecutor.shutdownNow();
```

Comment:
Just making sure: this line automatically cancels the scheduled runnable in `_scheduledRefreshFuture` right? If the refresh period is right after this line and before the next line, it's okay?

Reply:
Yes, the shutdownNow method will automatically cancel the scheduled tasks. Now I changed to another way for reset (not shutting down), I only cancel the scheduled future now, instead of shut down the executor and initialize another one, I think it's more efficient.